### ISSUE TITLE
Dataset class imbalance problem

### ISSUE THREAD

**Issue by nabilLearns** [nabilLearns/ChestXRay#1]
**Motivation: Need to convert a multilabel task into a multiclass task**
The ChestXRay classification task as I have currently framed it is a multilabel classification problem. Because of this, I cannot infer what class(es) an image belongs to from the logits output by a CNN (i.e. with softmax of logits). I can, however, do this in a multiclass classification problem. I can transform the multilabel into a multiclass problem with the powerset of the 15 possible labels an image can have. The cardinality of this set is huge, so I wanted to investigate how realistic it would be to tackle this multiclass problem with my current dataset (which is a subset of the total dataset, I've only downloaded ~2GB out of ~50GB of data).

**Multiclass formulation might not be realistic given the current dataset**
Below, is a histogram showing the distribution of classes. From this, we can clearly see there is a clear imbalance between the presence of classes in the dataset I currently have downloaded.

![image](https://user-images.githubusercontent.com/43799733/147887744-09bdb9be-be42-4475-80f2-ac03d1dccfc5.png)
